File: src/api/Api/Models/Employee.cs
namespace Api.Models;

public class Employee
{
    public int Id { get; set; }
    public required string FullName { get; set; }
    public required string Email { get; set; }
    public required string Position { get; set; }
    public required string Subdivision { get; set; }
    public required string Status { get; set; }
    public int? PeoplePartner { get; set; }
    public int OutOfOfficeBalance { get; set; }
    public required string Password { get; set; }
}

File: src/client/src/Types/Employee.ts
export interface Employee {
    id: number,
    fullName: string,
    email: string,
    position: string,
    subdivision: string,
    status: string,
    peoplePartner?: number,
    outOfOfficeBalance: number
}
export interface UserRegistration {
    id: number,
    fullName: string,
    email: string,
    position: string,
    subdivision: string,
    status: string,
    peoplePartner?: number,
    outOfOfficeBalance: number,
    password: string,
}

export interface UpdateUserInput {
    FullName: string,
    Email: string,
    Position: string,
    Subdivision: string,
    Status: string,
    PeoplePartner?: number,
    OutOfOfficeBalance: number
}

export interface CreateUserInput {
    FullName: string,
    Email: string,
    Position: string,
    Subdivision: string,
    Status: string,
    PeoplePartner?: number,
    OutOfOfficeBalance: number,
    Password: string
}